pkg/apis/operator/v1alpha1/types_garden.go

```diff
@@ -295,9 +309,15 @@ type Backup struct {
 	// Provider is a provider name. This field is immutable.
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Provider is immutable"
 	Provider string `json:"provider"`
-	// BucketName is the name of the backup bucket.
+	// BucketName is the name of the backup bucket. If not provided, gardener-operator attempts to manage a new bucket.
+	// In this case, the cloud provider credentials provided in the SecretRef must have enough privileges for creating
+	// and deleting buckets.
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="BucketName is immutable"
-	BucketName string `json:"bucketName"`
+	// +optional
+	BucketName *string `json:"bucketName,omitempty"`
```

Comment:
Does above validation rule make sure that you cannot set it if it was unset before?

Reply:
How can we migrate to "GOP managing backup bucket" for existing `Garden`s?

Reply:
I still see L340 is unchanged. Isn't this preventing above use-case?